library/src/main/java/bio/terra/pfb/PfbReader.java

```diff
@@ -48,23 +48,32 @@ public static String showMetadata(String fileLocation) throws IOException {
   }
 
   public static List<String> show(String fileLocation) throws IOException {
-    GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
-    URL url = isValidUrl(fileLocation);
-    GenericRecord genericRecord = null;
     List<String> data = new ArrayList<>();
-    try (InputStream in =
-            url != null ? readFromSignedUrl(url.toString()) : readFromLocalFile(fileLocation);
-        DataFileStream<GenericRecord> reader = new DataFileStream<>(in, datumReader)) {
-      // Skip Metadata Object, which should always appear first
-      reader.next(genericRecord);
-      while (reader.hasNext()) {
-        genericRecord = reader.next(genericRecord);
-        data.add(convertEnum(genericRecord.toString()));
+    try {
```

Comment:
This might be a good place to use `try with resources` on the `getGenericRecordsStream`, because we know we'll be done with the `DataFileStream` by the time we're done with this method so hopefully won't cause the same problems as doing it within `getGenericRecordsStream`.